commands/scanandfixrepos.go

```diff
@@ -33,12 +36,26 @@ func (saf *ScanAndFixRepositories) Run(configAggregator utils.FrogbotConfigAggre
 
 func (saf *ScanAndFixRepositories) scanAndFixSingleRepository(repoConfig *utils.FrogbotRepoConfig, client vcsclient.VcsClient) error {
 	for _, branch := range repoConfig.Branches {
-		err := saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		shouldScan, checkedCommit, err := saf.shouldScanLatestCommit(context.Background(), repoConfig, client, branch)
+		if err != nil {
+			return err
+		}
+		if !shouldScan {
+			log.Info(fmt.Sprintf("Commit %s has already been scanned by FrogBot", checkedCommit))
+			continue
+		}
+		err = saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		if err != nil {
+			errorDescription := fmt.Sprintf("FrogBot error %s", err)
+			log.Error(errorDescription)
+			err = saf.setCommitBuildStatus(client, repoConfig, vcsclient.Fail, checkedCommit, errorDescription)
+			return err
+		}
+		err = saf.setCommitBuildStatus(client, repoConfig, vcsclient.Pass, checkedCommit, "FrogBot scanned")
```

Comment:
You already created the `CommitStatusDescription` const, so you should probably remove the last argument for this function and use the const.